worker/draft.go

```diff
@@ -2038,6 +2038,21 @@ func (n *node) calculateSnapshot(startIdx, lastIdx, minPendingStart uint64) (*pb
 					maxCommitTs = x.Max(maxCommitTs, txn.CommitTs)
 				}
 			}
+
+			// If we encounter a restore proposal, we can immediately truncate the WAL and create
+			// a snapshot. This is to avoid the restore happening again if the server restarts.
+			if proposal.Restore != nil {
+				restoreTs := proposal.Restore.GetRestoreTs()
+				glog.Infof("Found restore proposal with restoreTs: %d", restoreTs)
```

Comment:
Add a log like `calculated snapshot from the restore proposal` and maybe dump the snapshot as well in the logs?